### PR TITLE
Update all dependencies to latest versions

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: bigquery/CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ### 1.2.1 - 2021-XX-XX
 
+1.  Update all dependencies to latest versions.
```
Comment (jaceklaskowski) [inline bigquery/CHANGES.md:3]: What do you think about making it a link to the PR? Since we're at it, why not.

File: pom.xml [hunk 2]
```diff
@@ -84,22 +84,22 @@
...
+    <google.api-bigquery.version>v2-rev20210303-${google.api-client-libraries.version}</google.api-bigquery.version>
     <google.api-iamcredentials.version>v1-rev20201022-${google.api-client-libraries.version}</google.api-iamcredentials.version>
-    <google.api-storage.version>v1-rev20201112-${google.api-client-libraries.version}</google.api-storage.version>
-    <google.auth.version>0.22.2</google.auth.version>
+    <google.api-storage.version>v1-rev20210127-${google.api-client-libraries.version}</google.api-storage.version>
+    <google.auth.version>0.24.1</google.auth.version>
     <google.auto-value.version>1.7.4</google.auto-value.version>
-    <google.cloud-bigquerystorage.version>1.7.2</google.cloud-bigquerystorage.version>
+    <google.cloud-bigquerystorage.version>1.15.0</google.cloud-bigquerystorage.version>
     <google.flogger.version>0.5.1</google.flogger.version>
```
Comment (suztomo) [inline pom.xml:96]: Would you you upgrade flogger's version as well?

https://github.com/google/flogger/releases/tag/flogger-0.6

Comment (suztomo) [reply pom.xml:96]: Once `com.google.cloud.bigdataoss:util` depends on newer version of flogger-system-backend, I can remove a workaround to avoid dependency conflict (https://github.com/apache/beam/pull/14578/commits/facd2c039dd1d2890d68ee48762bea271e368ad4#r617759024) from Apache Beam.

Comment (medb) [reply pom.xml]: Done.

File: bigquery/CHANGES.md (hunk 1 above)
Comment (medb) [reply bigquery/CHANGES.md]: What is the syntax for linking PR in `md` file?

File: bigquery/CHANGES.md [hunk 3]
```diff
@@ -1,543 +1,525 @@
 ### 1.2.1 - 2021-XX-XX
 
+1. Update all dependencies to latest versions.
```
Comment (mprashanthsagar) [inline bigquery/CHANGES.md:3]: Can we move all the cosmetic changes to a separate PR ?

**Review by mprashanthsagar — CHANGES_REQUESTED** [review]
Can we separate the PR into 3 
* Logger changes 
* Updated dependencies
* Checkstyle-changes

**Comment by mprashanthsagar** [comment]
Please address the build failure

File: bigquery/CHANGES.md (hunk 3 above)
Comment (medb) [reply bigquery/CHANGES.md]: Done.

**Comment by medb** [comment]
> Can we separate the PR into 3
> 
> * Logger changes
> * Updated dependencies
> * Checkstyle-changes

Reverted checkstyle changes - other changes are required for dependencies upgrade.

**Comment by medb** [comment]
> Please address the build failure

Done.

**Review by mprashanthsagar — APPROVED** [review]
